### PR TITLE
feat(shell): add app switcher styles

### PR DESCRIPTION
**Pull request by aledavila**
closes https://github.ibm.com/carbon/issues/issues/906

Preliminary styles for shell product switcher. 
Decided to split off functionality into another PR

To check to go to UI Shell under components

Staging Link:

UI Shell Default Expanded: 
http://shell-switcher-zany-tasmaniandevil.mybluemix.net/component/ui-shell--switcher-expanded-default

UI Shell All Products:
http://shell-switcher-zany-tasmaniandevil.mybluemix.net/component/ui-shell--switcher-all-products

### CODE REVIEW

File: src/globals/scss/_feature-flags.scss
Comment (alisonjoseph) [inline src/globals/scss/_feature-flags.scss]: make sure this gets switched back to false before merging

Comment (aledavila) [reply src/globals/scss/_feature-flags.scss]: yes! thanks for catching that

Comment (alisonjoseph) [reply src/globals/scss/_feature-flags.scss]: Only because @joshblack caught it in my PR 😉

**Comment by aledavila** [comment]
I'm sorry omg ok updated this

File: src/components/ui-shell/shell-switcher.hbs
Comment (joshblack) [inline src/components/ui-shell/shell-switcher.hbs]: Nit: `ariaLabel` -> `aria-label`

File: src/components/ui-shell/_shell-switcher.scss
Comment (joshblack) [inline src/components/ui-shell/_shell-switcher.scss]: Nit: prefer un-nesting for selectors themselves 👍

File: src/components/ui-shell/shell-switcher.hbs
Comment (joshblack) [inline src/components/ui-shell/shell-switcher.hbs]: For class naming, could it be more appropriate to refer to the `item` itself versus the list? e.g.

```
.bx--app-list__link-img {}
.bx--app-link__img {}
// etc.
```

**Review by joshblack — CHANGES_REQUESTED** [review]
Minor nitpicks, feel free to ignore them 😄 Should we also get Anna on as a design reviewer and share a staging link?

**Comment by aledavila** [comment]
@joshblack thanks will take a look

**Comment by joshblack** [comment]
Let me know if you need anything else @aledavila!

**Comment by aledavila** [comment]
@joshblack it should be all good now

**Comment by joshblack** [comment]
@aledavila could you create a staging link and then bring in a design reviewer, as well?

File: src/components/ui-shell/_shell-switcher.scss
Comment (joshblack) [inline src/components/ui-shell/_shell-switcher.scss]: Would you prefer using `units` or `get-fixed-size` for these kinds of things? I've been using the latter for some of the other recent work but am down to change back 👍

File: src/components/ui-shell/_shell-switcher.scss
Comment (joshblack) [inline src/components/ui-shell/_shell-switcher.scss]: Should we tie this into an ibm color?

File: src/components/ui-shell/_shell-switcher.scss
Comment (joshblack) [inline src/components/ui-shell/_shell-switcher.scss]: Ideally these should be in 4, 8, 16, etc. Is the 3px from the comp?

File: src/components/ui-shell/_shell-switcher.scss
Comment (joshblack) [inline src/components/ui-shell/_shell-switcher.scss]: The unit values should be whole numbers, right? Or am I missing something?

File: src/components/ui-shell/_shell-switcher.scss
Comment (joshblack) [inline src/components/ui-shell/_shell-switcher.scss]: Should we sync #000 with the black token from the ibm colors?

**Review by joshblack — CHANGES_REQUESTED** [review]
Just a couple of comments, looks great! Love the handlebars stuff you added, thank you for that 🙏

File: src/components/ui-shell/_shell-switcher.scss
Comment (aledavila) [reply src/components/ui-shell/_shell-switcher.scss]: This was because the outline on focus is 3px and it was only showing the outline on the top and bottom and not on the sides. should i remove this instead?

File: src/components/ui-shell/_shell-switcher.scss
Comment (aledavila) [reply src/components/ui-shell/_shell-switcher.scss]: It should yes but the subheader and link to "show all" weren't aligning with the list items and I wasn't sure how to go about it. Do you have any thoughts on this.

**Review by chrsfshr — COMMENTED** [review]
LGTM! Only thing that stood out to me was using the IBM color vars in some places but I think @joshblack covered all those!

File: src/components/ui-shell/_shell-switcher.scss
Comment (asudoh) [inline src/components/ui-shell/_shell-switcher.scss]: Can we tokenize the color here...?

File: src/components/ui-shell/_shell-switcher.scss
Comment (asudoh) [inline src/components/ui-shell/_shell-switcher.scss]: You may want to test this with IE11, where [`unset` is not supported](https://developer.mozilla.org/en-US/docs/Web/CSS/unset#Browser_compatibility).

**Review by asudoh — COMMENTED** [review]
Just a couple of comments from mine - Thanks @aledavila!

**Comment by aledavila** [comment]
@joshblack @asudoh @chrsfshr I've updated and added a staging link

**Review by asudoh — COMMENTED** [review]
LGTM 👍 once design team signs this off. Thanks @aledavila!

File: src/components/ui-shell/shell-switcher.hbs
Comment (aledavila) [inline src/components/ui-shell/shell-switcher.hbs]: Per Anna's request it is a subheader in the meantime since there is no functionality

File: src/components/ui-shell/shell-switcher.hbs
Comment (aledavila) [inline src/components/ui-shell/shell-switcher.hbs]: Per Anna's request it is a subheader in the meantime since there is no functionality

File: src/components/ui-shell/ui-shell.config.js
Comment (asudoh) [inline src/components/ui-shell/ui-shell.config.js]: @aledavila Did you really mean to remove `switcher` here? There still seems a [reference to `switcher`](https://github.com/IBM/carbon-components/blob/85865312fc3569a189fb82d9bbc9579d7d7a362f/src/components/ui-shell/ui-shell.config.js#L131), which seems a cause of [latest Travis failure](https://travis-ci.org/IBM/carbon-components/builds/430158583).

Comment (aledavila) [reply src/components/ui-shell/ui-shell.config.js]: It got removed in the merge accidentally have added it back now. Thanks

**Review by aagonzales — COMMENTED** [review]
- the overflow menu icon should only appear on hover/focus with each item.
- move "bx--app-link__name" over 8px to the right for type alignment.
- also I can we add the back button back to the "Switcher All Products" version. I didn't realize there were multiple versions the first time. That was my bad.

File: src/components/ui-shell/shell-switcher.hbs
Comment (joshblack) [inline src/components/ui-shell/shell-switcher.hbs]: Will the overflow menu work if it's in an `<a>` tag?

File: src/components/ui-shell/shell-switcher.hbs
Comment (joshblack) [inline src/components/ui-shell/shell-switcher.hbs]: Something that might help with the placeholder here could be to use the "stop" or "fade" icons from the Duo icon library. Here are the SVG assets:

Stop.svg

```svg
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32"><path d="M16 4A12 12 0 1 1 4 16 12.014 12.014 0 0 1 16 4m0-2a14 14 0 1 0 14 14A14 14 0 0 0 16 2z"/><path d="M20 12v8h-8v-8h8m0-2h-8a2 2 0 0 0-2 2v8a2 2 0 0 0 2 2h8a2 2 0 0 0 2-2v-8a2 2 0 0 0-2-2z"/></svg>
```

Fade.svg

```svg
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32"><path d="M8.24 25.14L7 26.67a14 14 0 0 0 4.18 2.44l.68-1.88a12 12 0 0 1-3.62-2.09zm-4.05-7.07l-2 .35A13.89 13.89 0 0 0 3.86 23l1.73-1a11.9 11.9 0 0 1-1.4-3.93zm7.63-13.31l-.68-1.88A14 14 0 0 0 7 5.33l1.24 1.53a12 12 0 0 1 3.58-2.1zM5.59 10L3.86 9a13.89 13.89 0 0 0-1.64 4.54l2 .35A11.9 11.9 0 0 1 5.59 10zM16 2v2a12 12 0 0 1 0 24v2a14 14 0 0 0 0-28z"/></svg>
```

What I've been doing on the other parts of the codebase is basically:

```scss
.#{$prefix}--some-icon-class {
  // Set width and height explicitly.
  // Icons seem to come in three sizes:
  // 48x48, 48x40, 48x32
}
```

And then in the markup it becomes:

```html
<div class="bx--some-icon-class">
  <svg width="20" height="20" aria-hidden="true" ...>
     ...
  </svg>
</div>
```

We might be able to refactor this out in the future, too, to have similar icon classes for everything.

File: src/components/ui-shell/shell-switcher.hbs
Comment (joshblack) [inline src/components/ui-shell/shell-switcher.hbs]: Could this just be:

```
.bx--app-switcher__search
```

**Review by joshblack — CHANGES_REQUESTED** [review]
Left a couple of comments. Might help at the top-level to keep using `bx--app-switcher` instead of `bx--switcher`, as well.

File: src/components/ui-shell/shell-switcher.hbs
Comment (aledavila) [reply src/components/ui-shell/shell-switcher.hbs]: I assumed since the overflow menu is a hover over it was fine. Do you have any suggestions on a different approach to this?

Comment (joshblack) [reply src/components/ui-shell/shell-switcher.hbs]: I think it depends on if the event is propagated? When I clicked on the overflow, it seemed to use the link which is why I was asking.

I think it gets hard if you need the hover to span the full width and you need the Overflow outside of the element 😞 initial thought was to just have the container be `position: relative` and position the overflow menu absolutely within the container but that can definitely get messy.

Maybe you can have them defined as siblings, but the hover color comes from hovering on the parent container?

Comment (aledavila) [reply src/components/ui-shell/shell-switcher.hbs]: I can try that. The overflow hover color different from the a hover. The only thing is that the overflow appears only when you hover over the a. But gonna try to see if I can make it work.

Comment (joshblack) [reply src/components/ui-shell/shell-switcher.hbs]: Maybe you could do something like:

```scss
a:hover ~ .overflow-menu,
a:focus ~ .overflow-menu {
}
```

?

**Review by joshblack — APPROVED** [review]

**Review by aagonzales — APPROVED** [review]

**Pull request merged**
